packages/next-typesafe-url/src/app.ts

```diff
@@ -13,7 +13,7 @@ import type {
   InferLayoutPropsType as ILPT,
   UseAppParamsResult,
   UseParamsResult,
-} from "./types";
+} from "./static-types";
 import { createElement, useRef } from "react";
 
 export type { DynamicLayout };
```

Comment:
adding to the type exports discussion, the exports here should be unique from the exports in index, whatever those end up being. 

maybe move the Infers to static-types and import?